src/test/java/uk/gov/ons/census/casesvc/messaging/SampleReceiverIT.java

```diff
@@ -69,16 +71,43 @@ public void testHappyPath() throws InterruptedException, IOException {
 
     String actualMessage = queue1.poll(10, TimeUnit.SECONDS);
     assertNotNull("Did not receive message before timeout", actualMessage);
-    CaseCreatedEvent caseCreatedEvent =
-        objectMapper.readValue(actualMessage, CaseCreatedEvent.class);
-    assertNotNull(caseCreatedEvent);
-    assertEquals("RM", caseCreatedEvent.getEvent().getChannel());
+    FanoutEvent fanoutEvent = objectMapper.readValue(actualMessage, FanoutEvent.class);
+    assertNotNull(fanoutEvent);
+    assertEquals("RM", fanoutEvent.getEvent().getChannel());
+
+    actualMessage = queue1.poll(10, TimeUnit.SECONDS);
+    assertNotNull("Did not receive message before timeout", actualMessage);
+    fanoutEvent = objectMapper.readValue(actualMessage, FanoutEvent.class);
+    assertNotNull(fanoutEvent);
+    assertEquals("RM", fanoutEvent.getEvent().getChannel());
 
     actualMessage = queue2.poll(10, TimeUnit.SECONDS);
     assertNotNull("Did not receive message before timeout", actualMessage);
-    caseCreatedEvent = objectMapper.readValue(actualMessage, CaseCreatedEvent.class);
-    assertNotNull(caseCreatedEvent);
-    assertEquals("RM", caseCreatedEvent.getEvent().getChannel());
+    fanoutEvent = objectMapper.readValue(actualMessage, FanoutEvent.class);
+    assertNotNull(fanoutEvent);
+
+    boolean haveSeenCaseCreatedEvent = false;
+    boolean haveSeenUacUpdatedEvent = false;
+
+    if (fanoutEvent.getEvent().getType() == EventType.CASE_CREATED) {
```

Comment:
Don't 100% get this code area.  If it is required to look at queue2 twice can this whole thing be stuck in a nice clean function that returns true false on the haveSeenCaseCreatedEvent/haveSeenUacUpdatedEvent .

This would make the test a lot more readable to me.

Reply:
have refactored